### ISSUE TITLE
When Brave is re-installed with the same binary, an update triggered, and `Relaunch` button shows up on `brave://settings/help` page when there are no updates available

### ISSUE THREAD

**Issue by MadhaviSeelam** [brave/brave-browser#23649]
## Description 

As part of the testing of Omaha update to 1.3.36.111, tested install and re-install with same version. During first install, `brave://settings/help` page shows correct version (without update trigger). However, after re-install of the same binary, `brave://settings/help` page shows update trigger with `Relaunch` button.  Fidler logs shows same omaha updater version 1.3.36.111 in both install and re-install.  Same issue with current public release 1.40.105 Chromium: 103.0.5060.53 which has older Omaha version, 1.3.361.113. 

## Steps to Reproduce

1.  Download `BraveBrowserStandaloneBetaSetup.exe` and install standalone binary (Beta v1.41.74 (Chromium 103.0.5060.53)) 
2.  Go to` brave://settings/help`
3.  Page renders `Brave is up to date` message (No update trigger)
4.   `Re-install` same binary
5.   Go to `brave://settings/help`

## Actual result:

 Nearly up to date! message shows with `Relaunch` button on the second install

![image](https://user-images.githubusercontent.com/98358127/175361011-40cfb905-485d-4971-b06d-fb4da4176982.png)

## Expected result:
Relaunch button should not be available if no updates available

## Reproduces how often: 

Easily

## Brave version (brave://version info)

Brave | 1.41.74 Chromium: 103.0.5060.53 (Official Build) beta (64-bit)
-- | --
Revision | a1711811edd74ff1cf2150f36ffa3b0dae40b17f-refs/branch-heads/5060@{#853}
OS | Windows 11 Version 21H2 (Build 22000.739)

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

@mherrmann

**Comment by GeetaSarvadnya** [comment]
@mherrmann Could you please look into this issue, the issue exists on both the new/old Omaha versions. When we install the standalone/stub/silent installers, we get the `brave is up to date` in brave:///settings/help, if we close and relaunch the browser then we get `Relaunch` in brave://settings/help although there is no new update. If we click on Relaunch we get the same existing version.

**Comment by mherrmann** [comment]
My suspicion is that this also happens for Chrome.

**Comment by GeetaSarvadnya** [comment]
@mherrmann Yup, you are right. I am seeing the same behavior in chrome as well. Looks like an upstream bug? 
![image](https://user-images.githubusercontent.com/38657976/176114671-971abf45-6893-4a04-8578-e13bc9daf2bf.png)

**Comment by mherrmann** [comment]
I would say so, yes.

**Comment by GeetaSarvadnya** [comment]
@mkarolin @simonhong: It's an upstream bug, shall we keep this bug open until chrome fixes the issue? defer to @rebron to take the decision

cc: @LaurenWags

**Comment by rebron** [comment]
@mherrmann Do you know what the upstream bug handy is? I'm not seeing it from this list `https://bugs.chromium.org/p/chromium/issues/list?sort=-opened&q=relaunch%20button&can=2`
